tests/unit/h/services/annotation_stats_test.py

```diff
@@ -1,160 +1,114 @@
+        Search.return_value.clear.assert_called_once_with()
+        Limiter.assert_called_once_with()
+        DeletedFilter.assert_called_once_with()
+        UserFilter.assert_called_once_with()
+        assert Search.return_value.append_modifier.call_args_list == [
+            call(Limiter.return_value),
+            call(DeletedFilter.return_value),
+            call(UserFilter.return_value),
+        ]
+        Search.return_value.run.assert_called_with(
+            {"limit": 0, "user": sentinel.userid}
+        )
+        assert num_annotations == Search.return_value.run.return_value.total
 
-    def test_group_annotation_count_excludes_replies(
-        self, svc, search, top_level_annotation_filter
+    def test_group_annotation_count(
+        self, pyramid_request, svc, Search, TopLevelAnnotationsFilter
     ):
-        svc.group_annotation_count("groupid")
+        num_annotations = svc.group_annotation_count(sentinel.pubid)
 
-        search.return_value.append_modifier.assert_called_with(
-            top_level_annotation_filter.return_value
+        Search.assert_called_with(pyramid_request)
+        TopLevelAnnotationsFilter.assert_called_once_with()
+        Search.return_value.append_modifier.assert_called_once_with(
+            TopLevelAnnotationsFilter.return_value
         )
+        Search.return_value.run.assert_called_with(
+            {"limit": 0, "group": sentinel.pubid}
+        )
+        assert num_annotations == Search.return_value.run.return_value.total
 
-    def test_group_annotation_count_returns_total(self, svc, search):
-        search.return_value.run.return_value.total = 3
-
-        anns = svc.group_annotation_count("groupid")
-
-        assert anns == 3
+    @pytest.fixture
+    def svc(self, pyramid_request):
+        return AnnotationStatsService(request=pyramid_request)
 
 
 class TestAnnotationStatsFactory:
-    def test_returns_service(self):
-        svc = annotation_stats_factory(mock.Mock(), mock.Mock())
-
-        assert isinstance(svc, AnnotationStatsService)
-
-    def test_sets_request(self):
-        request = mock.Mock()
-        svc = annotation_stats_factory(mock.Mock(), request)
-
-        assert svc.request == request
-
-
-@pytest.fixture
-def svc(pyramid_request):
-    return AnnotationStatsService(request=pyramid_request)
+    def test_it(self, AnnotationStatsService):
+        svc = annotation_stats_factory(sentinel.context, sentinel.request)
+
+        AnnotationStatsService.assert_called_once_with(sentinel.request)
+        assert svc == AnnotationStatsService.return_value
+
+    @pytest.fixture(autouse=True)
+    def AnnotationStatsService(self, mocker):
+        return mocker.patch(
+            "h.services.annotation_stats.AnnotationStatsService",
+            autospec=True,
+            spec_set=True,
+        )
 
 
-@pytest.fixture
-def search(patch):
-    return patch("h.services.annotation_stats.Search", autospec=Search, spec_set=True)
```

Comment:
All these patch fixtures were weird in a number of ways:

1. It's a patch for the `Search` class so the fixture should be called `Search` not `search`
2. You don't need to pass `autospec` to `patch()`, it [does it automatically](https://github.com/hypothesis/h/blob/facbef0672740155e3cdd328ff53221d9cc43ade/tests/unit/h/conftest.py#L48). Also, `autospec` is a boolean argument, you don't need to pass it the class that it the `Search` being mocked (which means you also don't need to import the `Search` class)
3. Patch fixtures should always have `autouse=True` (see [previous discussion](https://hypothes-is.slack.com/archives/C1MA4E9B9/p1718193955021929))
4. The fixtures were in the wrong order in the test file (should match the order of the imports in the source file)